src/testResultsFile.ts

```diff
@@ -0,0 +1,76 @@
+"use strict";
+import * as fs from "fs";
+import * as path from "path";
+import { setTimeout } from "timers";
+import { Disposable, Event, EventEmitter } from "vscode";
+import { DOMParser } from "xmldom";
+import { TestResult } from "./testResult";
+
+function getAttributeValue(node, name: string): string {
+    const attribute = node.attributes.getNamedItem(name);
+    return (attribute === null) ? null : attribute.nodeValue;
+}
+
+export class TestResultsFile implements Disposable {
+    private static readonly ResultsFileName = "TestExplorerResults.trx";
+    private onNewResultsEmitter = new EventEmitter<TestResult[]>();
+    private resultsFile: string;
+    private watcher: fs.FSWatcher;
+
+    public constructor() {
+        const tempFolder = process.platform === "win32" ? process.env.TEMP : "/tmp";
+        this.resultsFile = path.join(tempFolder, TestResultsFile.ResultsFileName);
+
+        // The change event gets called multiple times, so use a one-second
+        // delay before we read anything to avoid doing too much work
+        const me = this;
+        let timeout: NodeJS.Timer;
+        this.watcher = fs.watch(tempFolder, (eventType, fileName) => {
+            if ((fileName === TestResultsFile.ResultsFileName) && (eventType === "change")) {
+                if (!timeout) {
```

Comment:
I think you could use `clearTimeout` here

Reply:
I'm not sure that does the same thing, as it a timeout previously established by calling `setTimeout`, however, we're calling it in the callback so there's nothing to cancel. I'm setting it to null so that I can see if I've already triggered a wait or not. I can introduce another flag variable if that would make the code easier to follow?

Reply:
I think you could refer to https://github.com/formulahendry/vscode-docker-explorer/blob/master/src/dockerTreeBase.ts#L21:31 The logic should be similar.
In current implementation, if the change event lasts for 30 seconds continuously (say every 100ms), `parseResults` will be called 30 times.
With the `clearTimeout`, it would be called only once (if the change event is 100ms, or less then 1000ms).

When the test is running, I guess the trx file will be changed continuously. I am not sure whether the TRX will always be a valid XML and could be read normally during the test is running. If it is, I think current implementation is good, since we could have a quick update when test is running.

Reply:
Like it, done! :smile: